packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -364,21 +364,54 @@ export class PanelPlugin<
 
   /**
    * Sets function that can return visualization examples and suggestions.
-   * @alpha
+   * @deprecated use setSuggestionsHandler
    */
   setSuggestionsSupplier(supplier: VisualizationSuggestionsSupplier) {
     this.suggestionsSupplier = supplier;
     return this;
   }
 
   /**
-   * Returns the suggestions supplier
    * @alpha
+   * Sets function that returns visualization suggestions based on the current panel data summary
+   */
+  setSuggestionsHandler(handler: VisualizationSuggestionsHandler<TOptions, TFieldConfigOptions>) {
+    this.suggestionsSupplier = {
+      getSuggestionsForData: (builder) => {
+        const appender = builder.getListAppender<TOptions, TFieldConfigOptions>({
+          pluginId: this.meta.id,
+          name: this.meta.name,
+        });
+
+        const result = handler(builder.dataSummary);
+
+        if (Array.isArray(result)) {
+          appender.appendAll(result);
+        } else if (result === true) {
+          appender.append({});
+        }
+      },
+    };
+
+    return this;
+  }
+
+  /**
+   * Returns the suggestions supplier
+   * @deprecated use hasSuggestionsSupplier
    */
   getSuggestionsSupplier(): VisualizationSuggestionsSupplier | undefined {
     return this.suggestionsSupplier;
   }
 
+  /**
+   * @alpha
+   * returns whether the plugin has configured suggestions
+   */
+  hasSuggestions(): boolean {
```

Comment:
we want to use this method in `getAllSuggestions` to determine whether suggestions have been configured for this plugin